### PR TITLE
[ACS-7381][ADF] Break DataTable dependency on Material Module

### PR DESCRIPTION
**Pull request by tamaragruszka**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://hyland.atlassian.net/browse/ACS-7381

**What is the new behaviour?**
All components and directives of datatable are standalone

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts [hunk 1]
```diff
@@ -15,34 +15,23 @@
  * limitations under the License.
  */
 
-import { ChangeDetectionStrategy, Component, Input, OnInit, ViewEncapsulation, OnDestroy, inject } from '@angular/core';
+import { AsyncPipe, NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject, Input, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { ClipboardModule } from '../../../clipboard';
```
[the remark this answers is not in the document]
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts:22]: yeah, all those components have some connections between

File: lib/core/src/lib/datatable/datatable.module.ts [hunk 2]
```diff
@@ -102,8 +67,15 @@ import { IconCellComponent } from './components/icon-cell/icon-cell.component';
...
-        DropZoneDirective
+        BooleanCellComponent,
+        AmountCellComponent,
+        NumberCellComponent,
+        LocationCellComponent,
+        DateCellComponent,
+        LocalizedDatePipe,
+        IconCellComponent
     ],
+    declarations: [EmptyListHeaderDirective, EmptyListBodyDirective, EmptyListFooterDirective],
```
[the remark this answers is not in the document]
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/datatable.module.ts]: I missed that directives and didn't change them to standalone before

File: lib/core/src/lib/datatable/components/icon-cell/icon-cell.component.spec.ts [hunk 3]
```diff
@@ -18,8 +18,7 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { IconCellComponent } from './icon-cell.component';
-import { ObjectDataTableAdapter } from '../../data/object-datatable-adapter';
-import { ObjectDataColumn } from '../../data/object-datacolumn.model';
+import { ObjectDataTableAdapter, ObjectDataColumn } from '@alfresco/adf-core';
```
Comment (DenysVuika) [inline lib/core/src/lib/datatable/components/icon-cell/icon-cell.component.spec.ts:21]: incorrect import path, should not be importing from the itself via namespace

File: lib/core/src/lib/datatable/components/json-cell/json-cell.component.spec.ts [hunk 4]
```diff
@@ -15,15 +15,12 @@
...
  */
 
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { ObjectDataTableAdapter } from '../../data/object-datatable-adapter';
-import { ObjectDataColumn } from '../../data/object-datacolumn.model';
-import { CoreTestingModule } from '../../../testing/core.testing.module';
-import { JsonCellComponent } from './json-cell.component';
-import { TranslateModule } from '@ngx-translate/core';
-import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { CoreTestingModule, JsonCellComponent, ObjectDataColumn, ObjectDataTableAdapter } from '@alfresco/adf-core';
```
Comment (DenysVuika) [inline lib/core/src/lib/datatable/components/json-cell/json-cell.component.spec.ts:18]: this type of import is incorrect as might produce circular loops

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/core/src/lib/datatable/components/date-cell/date-cell.component.spec.ts [hunk 5]
```diff
@@ -15,16 +15,14 @@
...
 
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { DateCellComponent } from './date-cell.component';
-import { DataColumn, DateConfig } from '../../data/data-column.model';
-import { BehaviorSubject } from 'rxjs';
-import { AppConfigService } from '../../../app-config';
-import { HttpClientTestingModule } from '@angular/common/http/testing';
-import { TranslateModule } from '@ngx-translate/core';
-import { LOCALE_ID } from '@angular/core';
+import { AppConfigService, DataColumn, DateCellComponent, DateConfig } from '@alfresco/adf-core';
```
Comment (DenysVuika) [inline lib/core/src/lib/datatable/components/date-cell/date-cell.component.spec.ts:18]: incorrect imports

File: lib/core/src/lib/datatable/components/number-cell/number-cell.component.spec.ts [hunk 6]
```diff
@@ -15,13 +15,13 @@
  * limitations under the License.
  */
 
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { NumberCellComponent } from './number-cell.component';
-import { DecimalConfig } from '../../data/data-column.model';
-import { BehaviorSubject } from 'rxjs';
-import { LOCALE_ID } from '@angular/core';
+import { DecimalConfig } from '@alfresco/adf-core';
```
Comment (DenysVuika) [inline lib/core/src/lib/datatable/components/number-cell/number-cell.component.spec.ts:18]: incorrect import

File: lib/core/src/lib/datatable/data-column/data-column.component.spec.ts [hunk 7]
```diff
@@ -15,18 +15,14 @@
  * limitations under the License.
  */
 
-import { DataColumnComponent } from './data-column.component';
-import { CoreTestingModule } from '../../testing/core.testing.module';
-import { TranslateModule } from '@ngx-translate/core';
+import { CoreTestingModule, DataColumnComponent } from '@alfresco/adf-core';
```
Comment (DenysVuika) [inline lib/core/src/lib/datatable/data-column/data-column.component.spec.ts:18]: incorrect import

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/core/src/lib/datatable/components/icon-cell/icon-cell.component.spec.ts (hunk 3 above)
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/components/icon-cell/icon-cell.component.spec.ts]: done

File: lib/core/src/lib/datatable/components/json-cell/json-cell.component.spec.ts (hunk 4 above)
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/components/json-cell/json-cell.component.spec.ts]: done

File: lib/core/src/lib/datatable/components/date-cell/date-cell.component.spec.ts (hunk 5 above)
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/components/date-cell/date-cell.component.spec.ts]: done

File: lib/core/src/lib/datatable/data-column/data-column.component.spec.ts (hunk 7 above)
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/data-column/data-column.component.spec.ts]: done

File: lib/core/src/lib/datatable/components/location-cell/location-cell.component.spec.ts [hunk 8]
```diff
@@ -15,11 +15,8 @@
  * limitations under the License.
  */
 
+import { CoreTestingModule, LocationCellComponent, ObjectDataColumn, ObjectDataTableAdapter } from '@alfresco/adf-core';
```
[the remark this answers is not in the document]
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/components/location-cell/location-cell.component.spec.ts]: done

File: lib/core/src/lib/datatable/components/number-cell/number-cell.component.spec.ts (hunk 6 above)
Comment (tamaragruszka) [reply lib/core/src/lib/datatable/components/number-cell/number-cell.component.spec.ts]: done

**Comment by tamaragruszka** [comment]
merge conflicts in a loop

**Pull request closed without merging**